armadillo/src/main/java/org/molgenis/armadillo/controller/StorageController.java

```diff
@@ -0,0 +1,158 @@
+package org.molgenis.armadillo.controller;
+
+import static io.swagger.v3.oas.annotations.enums.SecuritySchemeIn.COOKIE;
+import static io.swagger.v3.oas.annotations.enums.SecuritySchemeIn.HEADER;
+import static io.swagger.v3.oas.annotations.enums.SecuritySchemeType.APIKEY;
+import static io.swagger.v3.oas.annotations.enums.SecuritySchemeType.HTTP;
+import static org.springframework.http.HttpHeaders.CONTENT_DISPOSITION;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.MediaType.APPLICATION_OCTET_STREAM;
+import static org.springframework.http.MediaType.MULTIPART_FORM_DATA_VALUE;
+import static org.springframework.http.ResponseEntity.noContent;
+import static org.springframework.http.ResponseEntity.notFound;
+import static org.springframework.web.bind.annotation.RequestMethod.HEAD;
+
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.annotations.info.Info;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.security.SecurityScheme;
+import java.io.IOException;
+import java.util.List;
+import org.molgenis.armadillo.exceptions.FileProcessingException;
+import org.molgenis.armadillo.storage.ArmadilloStorageService;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@OpenAPIDefinition(
+    info = @Info(title = "MOLGENIS Armadillo - storage API", version = "0.1.0"),
+    security = {
+      @SecurityRequirement(name = "JSESSIONID"),
+      @SecurityRequirement(name = "http"),
+      @SecurityRequirement(name = "jwt")
+    })
+@SecurityScheme(name = "JSESSIONID", in = COOKIE, type = APIKEY)
+@SecurityScheme(name = "http", in = HEADER, type = HTTP, scheme = "basic")
+@SecurityScheme(name = "jwt", in = HEADER, type = APIKEY)
+@RestController
+@Validated
+@PreAuthorize("hasRole('ROLE_SU')")
+@RequestMapping("storage")
```

Comment:
careful: if we change mappings used in DSI we must also create new release of ArmadilloDSI package (and all packages that depend on it)?

Do we worry about that?

Reply:
There are a few methods of which I first thought that they should be moved to the storage API. (The ones that are marked with TODO in `DataController`). However, these endpoints behave differently: they list things from a "datashield" point of view. For example, listing all tables you can access as a user across all projects you can access. The Storage API works on a "single-project" level.

We need to keep both APIs so I will remove the TODOs. I want to refactor the `ArmadilloStorageService` a bit (maybe split it up) to make this distinction more clear.

Reply:
good